<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
    <script>
        /* 
            try{
                尝试执行的代码
            }catch(error){
                error   错误对象,try括号中代码执行的异常信息
                补救代码
            }
            执行过程：
                1.先去执行try中的代码
                2.如果try中的代码执行正常，catch中的代码就不执行了
                3.如果try中的代码执行异常，直接执行catch中的代码进行补救，

            try_throw_catch
            throw   手动抛出出异常

            try{
                尝试执行的代码
                throw new Error("异常信息文本");
            }catch(error){
                error   错误对象,try括号中代码执行的异常信息
                补救代码
            }
        */
       //先显示没出错行，到出错行，跳到catch    添加新功能和后期维护时候用
       var num = 10;
       try{
           alert("异常信息之前");
          /*  throw new Error("这是演习，不要紧张");//分段检查 */
           alert("异常信息之后");
           //1.要求num是5
           if(num != 5){
               throw new Error("num的值不是5");
           }
       }catch(error){
            alert("补救代码，ERROR：" + error);
       }
    </script>
</head>
<body>
    
</body>
</html>